src/middleware/file-validation.ts

```
@@ -0,0 +1,186 @@
+  file: Express.Multer.File,
+  cb: multer.FileFilterCallback,
+): void {
+  const ext = path.extname(file.originalname).toLowerCase();
+  const mime = file.mimetype.toLowerCase();
+
+  if (!ALLOWED_EXTENSIONS.has(ext) || !ALLOWED_MIME_TYPES.has(mime)) {
+    return cb(
+      Object.assign(new Error('Invalid file type'), {
+        code: 'UNSUPPORTED_TYPE',
+      }),
+    );
```

Biome (2.4.6)

[error] The function should not return a value because its return type is void.

(lint/correctness/noVoidTypeReturn)